core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -218,8 +218,8 @@ public void testFromXContent() throws IOException {
 
         QueryBuilder newQuery = queryParserService.queryParser(testQuery.getName()).fromXContent(context);
         assertNotSame(newQuery, testQuery);
-        assertEquals("Queries should be equal", testQuery, newQuery);
-        assertEquals("Queries should have equal hashcodes", testQuery.hashCode(), newQuery.hashCode());
+        assertEquals(testQuery, newQuery);
+        assertEquals("Queries should have equal hashcodes: " + testQuery + " vs. " + newQuery, testQuery.hashCode(), newQuery.hashCode());
```

Comment:
I left a previous comment around this, I had removed this text cause for me it causes duplication of the error message. Do you see something different? Junit already prints out "expect: .... actual: .... " with this addition it would print the same stuff out twice, at least that is what I used to see